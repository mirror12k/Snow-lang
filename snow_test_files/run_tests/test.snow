

print 'hello world!'

print true, false, nil, 5, 'asdf'

local #n = 3
while n > 0
	print n
	n = n - 1

local #n
print n

local #n = 3
while true
	print n
	n = n - 1
	if n == 0
		last
else
	print 'will never print 1'


unless 1 + 1 == 3
	print 'logic still works'
else
	print 'logic is broken'


local #n = 5

if n == 4
	print 'n is 4'
elseif n == 5
	print 'n is 5'
else
	print 'n is some unknown value!'

until n == 10
	print n
	n = n + 1


until n == 10
	last
else
	print 'last not triggered'


until n == 11
	print 'triggering last'
	last
else
	print 'will never print'


global $s = 'global s'
do
	local $s = 'local s'
	print 'my s:', s
print 'my s:', s


function testfun
	print 'hello world from testfun'

testfun()
testfun()

function testarg $
	print 'got arg:', s

testarg 'hello'
testarg()
testarg 5, 4


function testarg2 var1, var2
	print 'testarg2:', var1, var2

testarg2 5, 4, 3
testarg2 5

