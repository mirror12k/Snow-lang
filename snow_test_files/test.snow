
// first comment

/*
	hello world
*/

global a,b


