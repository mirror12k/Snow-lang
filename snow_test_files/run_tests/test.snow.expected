hello world!
true	false	nil	5	asdf
3
2
1
nil
3
2
1
logic still works
n is 5
5
6
7
8
9
last not triggered
triggering last
my s:	local s
my s:	global s
hello world from testfun
hello world from testfun
got arg:	hello
got arg:	nil
got arg:	5
testarg2:	5	4
testarg2:	5	nil
t:	5	15	-20	nil
t2:	5	4	3	nil
nested:	1	4	7	2	5	8	3	6	9
table
1
2
3
-1
-2
-3
aaa	5
1	5
2	4
3	3
